core/src/actors/chain_manager/data_request.rs

```diff
@@ -23,6 +25,38 @@ pub struct DataRequestPool {
 }
 
 impl DataRequestPool {
+    /// Get all the reveals
+    pub fn get_all_reveals(
+        &self,
+        utxo: &UnspentOutputsPool,
+    ) -> Vec<(
+        (OutputPointer, DataRequestOutput),
+        Vec<(OutputPointer, RevealOutput)>,
+    )> {
+        let mut v = vec![];
+        //self.data_request_pool.iter().filter()
```

Comment:
Put `TODO` here for signaling potential refactor to turn this function into a single expression.